packages/i18n/src/create-i18n.js

```diff
@@ -181,6 +191,14 @@ export const createI18n = ( initialData, initialDomain, hooks ) => {
 		notifyListeners();
 	};
 
+	/** @type {ResetLocaleData} */
+	const resetLocaleData = ( data, domain ) => {
+		// Reset all current Tannin locale data .
+		tannin.data = {};
```

Comment:
When resetting the `Tannin` instance, we should also set `tannin.pluralForms = {}`. It is a per-domain cache of plural form data derived from `this.data`. See how the cache is used here: https://github.com/aduth/tannin/blob/master/packages/tannin/index.js#L134-L168

Reply:
Thanks, I wasn't aware of this caching. Updated the code and added some tests.